src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -57,13 +82,18 @@ public final void doGet(HttpServletRequest request, HttpServletResponse response
   @Override
   public final void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
-    Credential googleCredential = AuthenticationUtility.getGoogleCredential(request);
-    if (googleCredential == null) {
+    try {
+      Credential googleCredential = getGoogleCredential(request);
+      if (googleCredential == null) {
+        response.sendError(403, ERROR_403);
+        return;
+      }
+      doPost(request, response, googleCredential);
+    } catch (TokenVerificationError e) {
       response.sendError(403, ERROR_403);
-      return;
+    } catch (GeneralSecurityException | IOException e) {
+      response.sendError(500, e.getMessage());
```

Comment:
Same comment about catching generic exceptions as in `doGet`

Reply:
Fixed same way as above!